packages/benchmarks/src/textToDriver/nodeJsDriverBenchmark.ts

```diff
@@ -0,0 +1,167 @@
+import { runTextToDriverEval } from "./runTextToDriverEval";
+import { radiantModels } from "../radiantModels";
+import { assertEnvVars, MongoClient } from "mongodb-rag-core";
+import { NODE_JS_PROMPTS } from "./generateDriverCode/languagePrompts/nodeJs";
+import OpenAI, { AzureOpenAI } from "openai";
+import { TEXT_TO_DRIVER_ENV_VARS } from "./TextToDriverEnvVars";
+import { RADIANT_ENV_VARS } from "../envVars";
+import { wrapOpenAI } from "braintrust";
+import PromisePool from "@supercharge/promise-pool";
+async function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+async function main() {
+  const {
+    BRAINTRUST_API_KEY,
+    BRAINTRUST_TEXT_TO_DRIVER_PROJECT_NAME,
+    MONGODB_TEXT_TO_DRIVER_CONNECTION_URI,
+    RADIANT_API_KEY,
+    RADIANT_ENDPOINT,
+    MONGODB_AUTH_COOKIE,
+  } = assertEnvVars({ ...TEXT_TO_DRIVER_ENV_VARS, ...RADIANT_ENV_VARS });
+  const projectName = BRAINTRUST_TEXT_TO_DRIVER_PROJECT_NAME;
+  const datasetName = "text-to-query-results";
+  const maxConcurrency = 3;
+  const timeout = 60000;
+
+  const prompts = NODE_JS_PROMPTS.systemPrompts;
+  const mongoClient = new MongoClient(MONGODB_TEXT_TO_DRIVER_CONNECTION_URI);
+
+  const SAMPLE_DOCUMENT_LIMIT = 2;
+
+  try {
+    await mongoClient.connect();
+    await sleep(500);
```

Comment:
Curious what the sleep is for. Is there a potential race condition here (e.g. on a very slow network)?

Reply:
yeah, i was getting mongo errs w/o that, idk why..too many promises